ddtrace/ext/git.py

```diff
@@ -100,6 +100,20 @@ def extract_workspace_path(cwd=None):
     return workspace_path
 
 
+def extract_branch(cwd=None):
+    # type: (Optional[str]) -> str
+    """Extract git branch from the git repository in the current directory or one specified by ``cwd``."""
+    branch = _git_subprocess_cmd("rev-parse --abbrev-ref HEAD", cwd=cwd)
+    return branch
+
+
+def extract_commit_sha(cwd=None):
+    # type: (Optional[str]) -> str
+    """Extract git commit SHA from the git repository in the current directory or one specified by ``cwd``."""
+    commit_sha = _git_subprocess_cmd("rev-parse HEAD", cwd=cwd)
+    return commit_sha
```

Comment:
we've had some customers that were missing commit SHA and branch even though we could extract git metadata (they had commit message for example). This is because if we don't find these parameters in the CI env vars, we don't set them at all. This changes fixes this issue.